Framework/Muon/src/ConvertFitFunctionForMuonTFAsymmetry.cpp

```diff
@@ -379,4 +361,28 @@ ConvertFitFunctionForMuonTFAsymmetry::getTFAsymmFitFunction(const Mantid::API::I
 
   return std::dynamic_pointer_cast<IFunction>(multi);
 }
+
+void ConvertFitFunctionForMuonTFAsymmetry::configureMultiDomainFunction(
+    std::shared_ptr<MultiDomainFunction> multiDomainFunction, const Mantid::API::IFunction_sptr &userFunc,
+    const double normValue) {
+  auto constant = FunctionFactory::Instance().createInitialized("name = FlatBackground, A0 = 1.0, ties=(A0=1)");
+  auto inBrace = std::make_shared<CompositeFunction>();
+  inBrace->addFunction(constant);
+  inBrace->addFunction(userFunc);
+  auto norm = FunctionFactory::Instance().createInitialized("name = FlatBackground, A0 "
+                                                            "=" +
+                                                            std::to_string(normValue));
+  auto product = Kernel::DynamicPointerCastHelper::dynamicPointerCastWithCheck<CompositeFunction, API::IFunction>(
+      FunctionFactory::Instance().createFunction("ProductFunction"));
+  product->addFunction(norm);
+  product->addFunction(inBrace);
+  auto composite = Kernel::DynamicPointerCastHelper::dynamicPointerCastWithCheck<CompositeFunction, API::IFunction>(
```

Comment:
Just a question to help me - why is this necessary?
Note this line is also present in `PawleyFunction.cpp` and a few other places
https://github.com/mantidproject/mantid/blob/005e651f12b64e9dc38d472d5d74627d51a99af7/Framework/CurveFitting/src/Functions/PawleyFunction.cpp#L458

Reply:
If we do `dynamic_pointer_cast<T>` and then use the result without checking for null, then Coverity complains. In #38100 I added `dynamicPointerCastWithCheck` to include the null check before returning the pointer, (and save duplicating `if ( thing == nullptr)` in several places.